pkg/apis/ark/v1/backup.go

```diff
@@ -49,8 +49,40 @@ type BackupSpec struct {
 	// TTL is a time.Duration-parseable string describing how long
 	// the Backup should be retained for.
 	TTL metav1.Duration `json:"ttl"`
+
+	Hooks BackupHooks `json:"hooks"`
+}
+
+type BackupHooks struct {
+	Resources []BackupResourceHookSpec `json:"resources"`
```

Comment:
The reason I went with resources is because I want to distinguish between different types of hooks:
1. Those that we invoke when backing up a specific resource, such as a pod or deployment
1. Those that we invoke at the backup level, e.g. just before we start backing up resources, and then at the end, when we're done backing up all the resources

Here is a sample backup I've been testing:
```yaml
apiVersion: ark.heptio.com/v1
kind: Backup
metadata:
  generateName: b6-
  namespace: heptio-ark
spec:
  hooks:
    resources:
      -
        namespaces:
          - heptio-ark
        resources:
          - pods
        labelSelector:
          matchLabels:
            pod-template-hash: "2793807833"
        hooks:
          - exec:
              container: ark
              command:
                - /bin/uname
                - -a
              onError: Fail
              timeout: 1m
  includedNamespaces:
  - heptio-ark
  includedResources:
  - pods
  labelSelector:
    matchLabels:
      pod-template-hash: "2793807833"
  ttl: 24h0m0s
```

I don't think `HookSpecs` would be good, because then we'd have `spec.hooks.hookSpecs`. I'm open to other names, but I wanted something that "read" well in yaml.